base/serialize.jl

```diff
@@ -476,13 +536,14 @@ function deserialize(s, ::Type{Array})
     return A
 end
 
-deserialize(s, ::Type{Expr})     = deserialize_expr(s, int32(read(s, UInt8)))
-deserialize(s, ::Type{LongExpr}) = deserialize_expr(s, read(s, Int32))
+deserialize(s, ::Type{Expr})     = deserialize_expr(s, int32(read(s.io, UInt8)))
+deserialize(s, ::Type{LongExpr}) = deserialize_expr(s, read(s.io, Int32))
```

Comment:
Would be nice to have annotations on these since this will fail if `s` is not a `Serializer` object.